src/polymer/call-super-in-callbacks.ts

```diff
@@ -0,0 +1,221 @@
+          // Construct a nice legible warning.
+          const parsedDocumentContaining =
+              getParsedDocumentContaining(element.sourceRange, document);
+          if (parsedDocumentContaining) {
+            const correctSyntax = method.kind === 'constructor' ?
+                'super()' :
+                `super.${methodName}()`;
+            const nameOfMethod = method.kind === 'constructor' ?
+                'constructor' :
+                `${methodName} method override`;
+            warnings.push({
+              code: this.code,
+              severity: Severity.ERROR,
+              sourceRange:
+                  parsedDocumentContaining.sourceRangeForNode(method.key)!,
+              message: stripWhitespace(`
+                Elements that extend Polymer.Element must call
+                ${correctSyntax}
+                in their ${nameOfMethod}.
+            `)
+            });
+          }
+        }
+      }
+    }
+    return warnings;
+  }
+}
+
+function getParsedDocumentContaining(
+    sourceRange: SourceRange|undefined,
+    document: Document): ParsedDocument<any, any>|undefined {
+  if (!sourceRange) {
+    return undefined;
+  }
+  let mostSpecificDocument: undefined|Document = undefined;
+  for (const doc of document.getByKind('document')) {
+    if (isPositionInsideRange(sourceRange.start, doc.sourceRange)) {
+      if (!mostSpecificDocument ||
+          isPositionInsideRange(
+              doc.sourceRange!.start, mostSpecificDocument.sourceRange)) {
+        mostSpecificDocument = doc;
+      }
+    }
+  }
+  mostSpecificDocument = mostSpecificDocument || document;
+  return mostSpecificDocument.parsedDocument;
+}
+
+function getClassBody(astNode?: estree.Node|null): undefined|estree.ClassBody {
+  if (!astNode || !astNode.type) {
+    return undefined;
+  }
+  let classBody: undefined|estree.ClassBody = undefined;
+  estraverse.traverse(astNode, {
+    enter(node: estree.Node) {
+      if (node.type === 'ClassDeclaration' || node.type === 'ClassExpression') {
+        classBody = node.body;
+        return estraverse.VisitorOption.Break;
+      }
+    }
+  });
+  return classBody;
+}
+
+function mustCallSuper(
+    element: Element, methodName: string, document: Document): boolean {
+  // TODO(rictic): look up the inheritance graph for a jsdoc tag that describes
+  //     the method as needing to be called?
+  if (!methodsThatMustCallSuper.has(methodName)) {
+    return false;
+  }
+  // Did the element's super class define the method?
+  if (element.superClass) {
+    const superElement =
+        onlyOrNone(document.getById('element', element.superClass.identifier));
+    if (definesMethod(superElement, methodName)) {
+      return true;
+    }
+  }
+
+  return anyMixinDefinesMethod(element, methodName, document, true);
+}
+
+function doesCallSuper(method: estree.MethodDefinition, methodName: string) {
+  const statements = method.value.body.body;
+  const superCallTargets = statements.map((s) => {
```

Comment:
It looks like this only finds top level calls. As a test for a mixin that conditionally calls the super method.